### PR TITLE
Techiva Bloggers: data model

### PR DESCRIPTION
**Pull request by fobadara**
The following instructions were followed for this milestone:

Be sure to reference the [ERD diagram](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png) from the sneak peek if you have any questions about table or column names or data relationships.

- Build your project schema.
    - Create and run the necessary migration files.
    - Table and column names should match the ERD diagram.
        - note: photo for users table should be a link to an image
    - Foreign keys should be included.
    - All columns that are foreign keys should have a corresponding index.

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+  create_table "comments", force: :cascade do |t|
+    t.string "text"
+    t.datetime "created_at", precision: 6, null: false
+    t.datetime "updated_at", precision: 6, null: false
+    t.bigint "post_id", null: false
+    t.bigint "user_id", null: false
+    t.index ["post_id"], name: "index_comments_on_post_id"
+    t.index ["user_id"], name: "index_comments_on_user_id"
+  end
```
Comment (ahmetbozaci) [inline db/schema.rb]: - [ ] Table and column names should match the ERD diagram.  It should be author_id
[Requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/creating_data_model_project.md#:~:text=Table%20and%20column%20names%20should%20match%20the%20ERD%20diagram.)

File: db/schema.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+  create_table "likes", force: :cascade do |t|
+    t.datetime "created_at", precision: 6, null: false
+    t.datetime "updated_at", precision: 6, null: false
+    t.bigint "user_id", null: false
+    t.bigint "post_id", null: false
+    t.index ["post_id"], name: "index_likes_on_post_id"
+    t.index ["user_id"], name: "index_likes_on_user_id"
+  end
```
Comment (ahmetbozaci) [inline db/schema.rb]: - [ ] Table and column names should match the ERD diagram.  It should be author_id
[Requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/creating_data_model_project.md#:~:text=Table%20and%20column%20names%20should%20match%20the%20ERD%20diagram.)

File: db/schema.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+  create_table "posts", force: :cascade do |t|
+    t.text "title"
+    t.string "text"
+    t.integer "comments_counter"
+    t.integer "likes_counter"
+    t.datetime "created_at", precision: 6, null: false
+    t.datetime "updated_at", precision: 6, null: false
+    t.bigint "user_id", null: false
+    t.index ["user_id"], name: "index_posts_on_user_id"
```
Comment (ahmetbozaci) [inline db/schema.rb]: - [ ] Table and column names should match the ERD diagram.  It should be author_id
[Requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/creating_data_model_project.md#:~:text=Table%20and%20column%20names%20should%20match%20the%20ERD%20diagram.)

File: db/schema.rb [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  end
+
+  create_table "users", force: :cascade do |t|
+    t.datetime "created_at", precision: 6, null: false
+    t.datetime "updated_at", precision: 6, null: false
+  end
```
Comment (ahmetbozaci) [inline db/schema.rb:51]: - [ ] Users' table should include name, photo and bio columns. Please add it.

File: db/migrate/20220505142007_create_users.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+class CreateUsers < ActiveRecord::Migration[6.1]
+  def change
+    create_table :users do |t|
+      t.string :name
+      t.string :photo
+      t.text :bio
+      t.integereger :posts_counter
```
Comment (ahmetbozaci) [inline db/migrate/20220505142007_create_users.rb]: - [ ] It should be an `integer`. Please fix it.

File: db/migrate/20220505152424_create_comments.rb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+class CreateComments < ActiveRecord::Migration[6.1]
+  def change
+    create_table :comments do |t|
+      t.string :text
```
Comment (ahmetbozaci) [inline db/migrate/20220505152424_create_comments.rb]: The text type should be a `text`. Please change it.

File: db/migrate/20220505152800_create_post.rb [hunk 7]
```diff
@@ -0,0 +1,12 @@
+class CreatePost < ActiveRecord::Migration[6.1]
+  def change
+    create_table :posts do |t|
+      t.text :title
+      t.string :text
```
Comment (ahmetbozaci) [inline db/migrate/20220505152800_create_post.rb]: - [ ] The title type should be a string, text type should be a text. Please change it.

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# STATUS: Changes Required  ♻️


 @fobadara  ✋🏼 

#### Your project is great. You are almost done. However, there are some issues you should fix. Please fix them then your project will be approved.



### ♻️ Request changes ♻️

- [ ] _Check the comments under the review._

___

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
___

### Cheers and Happy coding! 💻 👏
almost


___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._


___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED 🍾✔🎉**
Hello @fobadara 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- You documented your work professionally.✔
- Github inline comments feature is used properly.
- You asked for the changes in a polite and friendly way.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**